### PR TITLE
report pytest coverage to codecov

### PR DESCRIPTION
**Pull request by mlin**
Coverage report: https://app.codecov.io/github/single-cell-data/TileDB-SOMA/tree/mlin%2Fcodecov/apis/python/src/tiledbsoma

I think we may be missing one step to authorize Codecov in the single-cell-data github org so that it can post comments+checks to PRs. (I think I'm a member of this repo but not the org)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -54,7 +54,11 @@ jobs:
       run: clang --version
 
     - name: Run pytests
-      run: python -m pytest apis/python/tests
+      run: PYTHONPATH=$(pwd)/apis/python/src python -m pytest --cov=apis/python/src --cov-report=xml apis/python/tests
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply .github/workflows/ci.yml:57]: @mlin awesome! Please say precisely that as a comment in the `ci.yml` file :)

[the remark this answers is not in the document]
Comment (johnkerl) [reply .github/workflows/ci.yml:57]: @mlin awesome! Please say precisely that as a comment in the ci.yml file :)

**Review by johnkerl — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -54,7 +54,16 @@ jobs:
...
+      # Setting PYTHONPATH ensures the tests load the in-tree source code unde apis/python/src
+      # instead of copy we `pip install`ed to site-packages above. That's needed for the code
+      # coverage analysis to work.
+      run: PYTHONPATH=$(pwd)/apis/python/src python -m pytest --cov=apis/python/src --cov-report=xml apis/python/tests
+
+    - name: Report coverage to Codecov
+      if: matrix.os == 'ubuntu-22.04' && matrix.python-version == '3.9'
+      uses: codecov/codecov-action@v3
+      with:
+        flags: python
```
Comment (mlin) [inline .github/workflows/ci.yml:66]: @eddelbuettel

Comment (eddelbuettel) [reply .github/workflows/ci.yml:66]: Yay, that should do the trick from what I have seen.

**Review by eddelbuettel — APPROVED** [review]

**Comment by mlin** [comment]
@eddelbuettel Actually, this seems to have failed the codecov action...

image

Comparing [this build](https://github.com/single-cell-data/TileDB-SOMA/actions/runs/3372431172/jobs/5595847257) to [the prior one](https://github.com/single-cell-data/TileDB-SOMA/actions/runs/3368031744/jobs/5586120576).

Looking into it. Just to confirm, did you authorize Codecov for the org or do we still need somebody to take care of that? Thanks

**Comment by eddelbuettel** [comment]
AFAIK I have no admin rights for the repo so I have not done anything.  As discussed I picked possible use of 'flags' from both the codecov website and the available options for the R function running and uploading coverage results. But all this are at the moment just educated guesses.

**Comment by mlin** [comment]
Update: the Codecov problem wasn't related to the flags, but in general Codecov has rate-limiting problems with unauthenticated use by public repos (relevant issues linked below). To solve this, we'll need an admin to furthermore create a Codecov auth token and put it in the repo secrets, and then I can modify the workflow to [pass the token to the Codecov action](https://github.com/codecov/codecov-action/issues/557#issuecomment-1224970469).

https://github.com/codecov/codecov-action/issues/837
https://github.com/codecov/codecov-action/issues/557

**Review by johnkerl — APPROVED** [review]

**Review by johnkerl — APPROVED** [review]

**Pull request merged**